static_src/stores/env_store.js

```diff
@@ -0,0 +1,58 @@
+import BaseStore from './base_store';
+import cfApi from '../util/cf_api';
+import { appActionTypes, envActionTypes } from '../constants';
+
+class EnvStore extends BaseStore {
+  constructor() {
+    super();
+    this.env = {};
+    this.updateError = {};
+    this.subscribe(() => this._registerToActions.bind(this));
+  }
+
+  getEnv(appGuid) {
+    return this.env[appGuid];
+  }
+
+  getUpdateError(appGuid) {
+    const error = this.updateError[appGuid];
+    if (!error) {
+      return null;
+    }
+    const { error_code: errorCode } = error;
+    return {
+      ...error,
+      errorCode
+    };
+  }
+
+  _registerToActions(action) {
+    switch (action.type) {
+      case envActionTypes.ENV_FOR_APP_FETCH:
+        cfApi.fetchEnv(action.appGuid);
```

Comment:
In general, we try to follow these guidelines for our stores: https://gist.github.com/twokul/9501770. 

We definitely have a few places where the code doesn't match this guide, and we are actively working to make those areas of the codebase follow the linked conventions!

Reply:
@el-mapache I've had a read of that gist, but I'm not sure how it relates to the line of code you're referencing - it may just be me, but the gist talks about V8 internals but it doesn't seem to link it back to how one should write one's JS code so V8 is happier?

Can you give me a little more info?

Reply:
Ha sorry! That was in my copy buffer, totally unrelated to this work. Here is the actual link: https://github.com/18F/cg-dashboard/blob/master/CONTRIBUTING.md#working-with-reactflux

This will make a lot more sense in context.

Reply:
Thanks, that makes more sense :) I've refactored the store. I think it's more along the lines of what you wanted and I am much more happy with it. I can see room for extracting some of this out as it could become a bit too boilerplate-y, but for now I think it's OK. See my commit message for other changes I made when I refactored the store.